agent/src/structures/mod.rs

```diff
@@ -4,29 +4,56 @@ use tokio::sync::Mutex;
 use utils::{blobs::BlobStore, errors::ServalError};
 use uuid::Uuid;
 
+use std::fs;
 use std::sync::Arc;
 use std::{collections::HashMap, path::PathBuf};
 
 /// Our application state. Fields are public for now but we'll want to fix that.
 #[derive(Debug, Clone, Serialize)]
 pub struct RunnerState {
     pub instance_id: Uuid,
+    pub extensions: HashMap<String, PathBuf>,
     pub storage: Option<BlobStore>,
     pub jobs: HashMap<String, JobMetadata>,
     pub total: usize,
     pub errors: usize,
 }
 
 impl RunnerState {
-    pub fn new(instance_id: Uuid, blob_path: Option<PathBuf>) -> Result<Self, ServalError> {
+    pub fn new(
+        instance_id: Uuid,
+        blob_path: Option<PathBuf>,
+        extensions_path: Option<PathBuf>,
+    ) -> Result<Self, ServalError> {
         let storage = match blob_path {
             Some(path) => Some(BlobStore::new(path)?),
             None => None,
         };
+        let extensions = if let Some(extensions_path) = extensions_path {
+            // Read the contents of the directory at the given path and build a HashMap that maps
```

Comment:
Since we enumerate the contents of the directory when the agent starts up, that means you can't hot-swap in new modules while the agent is running (although you can replace existing ones with new versions).

I've implemented it like this to avoid needing to do the work to enumerate available modules for every single job that runs. If the hot-swapping ends up being a problem, we can add a file watcher and/or a signal handler to force a re-scan.